src/query/server/server.go

```diff
@@ -608,13 +608,33 @@ func newStorages(
 ) (storage.Storage, cleanupFn, error) {
 	cleanup := func() error { return nil }
 
-	localStorage := m3.NewStorage(
+	// Setup query conversion cache.
+	var (
+		conversionCacheConfig = cfg.Cache.QueryConversionCacheConfiguration()
+		conversionCacheSize   = conversionCacheConfig.SizeOrDefault()
+	)
+
+	if err := conversionCacheConfig.Validate(); err != nil {
```

Comment:
Should validate prior to calling SizeOrDefault here

Reply:
then we should just check in `SizeOrDefault()`. having a validate doesn't make much sense unless we need to validate multiple things

Reply:
The benefit of validating like this is when we add config options to the validation method, won't have to worry about it being called in time, since it's already validated on startup